### PR TITLE
Fix check for ordering of left and right views in DateRangePicker

### LINKED ISSUE #785: DateRangePicker leftView doesnt update correctly when changing the rightView to be out of order
### Bug report

- __Package version(s)__: latest
- there is a bug in packages/datetime/src/dateRangePicker.tsx at line 557: function updateRightView,should be:
```javascript
        if (!leftView.isBefore(rightView)) {
             leftView = rightView.getPreviousMonth();
         }
```

### PR DESCRIPTION
**Pull request by chux0519**

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @chux0519! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: packages/datetime/src/dateRangePicker.tsx [hunk 1]
```diff
@@ -554,7 +554,8 @@ export class DateRangePicker
 
     private updateRightView(rightView: MonthAndYear) {
         let leftView = this.state.leftView.clone();
-        if (!rightView.isBefore(rightView)) {
+        // a bug here
+        if (!leftView.isBefore(rightView)) {
```
Comment (cmslewis) [inline packages/datetime/src/dateRangePicker.tsx]: @leebyp the original logic here does look funky. Can you take a look and verify this?

Also @chux0519, you can remove the `// a bug here` comment, won't be necessary or helpful after the PR merges.

**Comment by adidahiya** [comment]
@chux0519 please revert the changes to docs.js and write a more descriptive PR title.

File: packages/datetime/src/dateRangePicker.tsx (hunk 1 above)
Comment (leebyp) [inline packages/datetime/src/dateRangePicker.tsx]: @chux0519 there's a pattern for the checks here, can we change to `!rightView.isAfter(leftView)`?

even better would be a helper function of `viewsInOrder(leftView, rightView)`

**Comment by giladgray** [comment]
@chux0519 files in `docs/` directory should not be edited by human hands. we have build tasks that recompile that directory at relesae time.

File: packages/datetime/src/dateRangePicker.tsx (hunk 1 above)
Comment (chux0519) [inline packages/datetime/src/dateRangePicker.tsx]: ok sure, thanks for changing the title.

**Comment by leebyp** [comment]
@chux0519 thanks for the fix and merged!

**Pull request merged**